### PR TITLE
QModManager 3.2.1

### PR DESCRIPTION
**Pull request by MrPurple6411**
### Changes in Pull Request
- Main menu critical message service.
- Removed unnecessary file counting for improved speed.
- Fixed `NullReferenceException` on incorrect Entry Path.
- Dynamic Dialog Text handling for differences between Subnautica and BelowZero.
- Built using Release Configuration with Optimize Code enabled.
 

[Release file](https://github.com/SubnauticaModding/QModManager/files/4567093/QModManager_Setup.zip)

### CODE REVIEW

File: QModManager/Patching/ManifestValidator.cs [hunk 1]
```diff
@@ -202,6 +203,7 @@ internal ModStatus FindPatchMethods(QMod qMod)
                                 else
                                 {
                                     qMod.PatchMethods[patch.PatchOrder] = new QModPatchMethod(method, qMod, patch.PatchOrder);
+                                    continue;
```
Comment (PrimeSonic) [inline QModManager/Patching/ManifestValidator.cs:206]: A `continue` at the end of a loop block is redundant.

File: Executable/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -12,5 +12,5 @@
 
 [assembly: ComVisible(false)]
 
-[assembly: AssemblyVersion("3.1")]
-[assembly: AssemblyFileVersion("3.1")]
+[assembly: AssemblyVersion("3.2.0")]
+[assembly: AssemblyFileVersion("3.2.0")]
```
Comment (AlexejheroYTB) [inline Executable/Properties/AssemblyInfo.cs:16]:
```suggestion
[assembly: AssemblyVersion("3.2.0.0")]
[assembly: AssemblyFileVersion("3.2.0.0")]
```

File: Installer/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -14,5 +14,5 @@
 
 [assembly: Guid("8c6c9a0b-80c4-43d2-89f2-749e6f09fdda")]
 
-[assembly: AssemblyVersion("3.1")]
-[assembly: AssemblyFileVersion("3.1")]
+[assembly: AssemblyVersion("3.2")]
+[assembly: AssemblyFileVersion("3.2")]
```
Comment (AlexejheroYTB) [inline Installer/Properties/AssemblyInfo.cs:18]:
```suggestion
[assembly: AssemblyVersion("3.2.0.0")]
[assembly: AssemblyFileVersion("3.2.0.0")]
```

File: QModManager/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -13,8 +13,8 @@
 
 [assembly: ComVisible(false)]
 
-[assembly: AssemblyVersion("3.1")]
-[assembly: AssemblyFileVersion("3.1")]
+[assembly: AssemblyVersion("3.2.0")]
+[assembly: AssemblyFileVersion("3.2.0")]
```
Comment (AlexejheroYTB) [inline QModManager/Properties/AssemblyInfo.cs:17]:
```suggestion
[assembly: AssemblyVersion("3.2.0.0")]
[assembly: AssemblyFileVersion("3.2.0.0")]
```

**Comment by AlexejheroYTB** [comment]
Just one more thing before this PR is merged, a bug fix is needed.
A badly formatted mod.json causes an unhandled exception in QMM (instead of the mod failing to load, qmm itself breaks)

File: QModManager/Patching/ManifestValidator.cs (hunk 1 above)
Comment (MrPurple6411) [reply QModManager/Patching/ManifestValidator.cs:206]: https://discordapp.com/channels/324207629784186882/529634427110948899/706907934164123698

**Comment by MrPurple6411** [comment]
> Just one more thing before the next version is released, a bug fix is needed.
> A badly formatted mod.json causes an unhandled exception in QMM (instead of the mod failing to load, qmm itself breaks)

badly formed in what way... improper json or just missing required values?

**Comment by PrimeSonic** [comment]
@AlexejheroYTB @MrPurple6411 Version 3.2.0 is already available to the public as "open beta".
If you are going to continue making changes, update the version also.

**Pull request merged**